spring-cloud-contract-tools/spring-cloud-contract-gradle-plugin/src/main/groovy/org/springframework/cloud/contract/verifier/plugin/ContractsCopyTask.java

```diff
@@ -0,0 +1,414 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.contract.verifier.plugin;
+
+import java.io.File;
+
+import javax.inject.Inject;
+
+import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.file.DirectoryProperty;
+import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.provider.MapProperty;
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputDirectory;
+import org.gradle.api.tasks.Internal;
+import org.gradle.api.tasks.Nested;
+import org.gradle.api.tasks.Optional;
+import org.gradle.api.tasks.OutputDirectory;
+import org.gradle.api.tasks.PathSensitive;
+import org.gradle.api.tasks.PathSensitivity;
+import org.gradle.api.tasks.TaskAction;
+import org.springframework.cloud.contract.stubrunner.ContractDownloader;
+import org.springframework.cloud.contract.stubrunner.StubConfiguration;
+import org.springframework.cloud.contract.stubrunner.StubDownloader;
+import org.springframework.cloud.contract.stubrunner.StubDownloaderBuilderProvider;
+import org.springframework.cloud.contract.stubrunner.StubRunnerOptions;
+import org.springframework.cloud.contract.stubrunner.StubRunnerOptionsBuilder;
+import org.springframework.cloud.contract.stubrunner.spring.StubRunnerProperties;
+import org.springframework.cloud.contract.verifier.converter.ToYamlConverter;
+import org.springframework.util.StringUtils;
+
+// TODO: Convert to incremental task: https://docs.gradle.org/current/userguide/custom_tasks.html#incremental_tasks
+/**
+ * Task that copies the contracts in order for the jar task to
+ * generate the jar. It takes into consideration the inclusion
+ * patterns when working with repo with shared contracts.
+ *
+ * @author Marcin Grzejszczak
+ * @author Anatoliy Balakirev
+ * @author Shannon Pamperl
+ * @since 1.0.2
+ */
+class ContractsCopyTask extends DefaultTask {
+
+	static final String TASK_NAME = "copyContracts";
+	static final String CONTRACTS = "contracts";
+	static final String BACKUP = "original";
+
+	// inputs
+	private final Property<Boolean> convertToYaml;
+
+	private final Property<Boolean> failOnNoContracts;
+
+	private final DirectoryProperty contractsDirectory;
+
+	private final Dependency contractDependency;
+
+	private final Repository contractRepository;
+
+	private final Property<StubRunnerProperties.StubsMode> contractsMode;
+
+	private final MapProperty<String, String> contractsProperties;
+
+	private final Property<String> contractsPath;
+
+	// FIXME: I don't think this makes sense given the default configuration... This would only make sense if contractDir == project.projectDir
+	private final Property<Boolean> excludeBuildFolders;
+
+	// FIXME: deleting the stubs after tests breaks incremental builds
+	private final Property<Boolean> deleteStubsAfterTest;
```

Comment:
So I reviewed this one more thoroughly and found it's usage to be safe from the original Gradle up-to-date checks. The premise is that it's purging the cloned Git repository or Maven dependencies from a temporary folder.

I think it's name could definitely be better or we could simply just configure it to true and not have the parameter. 

Thoughts, @marcingrzejszczak?

Reply:
Since we're doing a breaking change release we can rename it and indeed set it by default to `true`. What name would you suggest?